src/System/reachability.rs

```diff
@@ -0,0 +1,156 @@
+use edbm::zones::OwnedFederation;
+
+use crate::ModelObjects::component::{State, Transition};
+use crate::TransitionSystems::{TransitionSystem, LocationID};
+use std::collections::HashMap;
+
+pub struct Path {
+    //start_state: State,
+    //transition: Transition,
+}
+
+pub fn preliminary_check(
+    start_state: &State,
+    end_state: &State,
+    system: &dyn TransitionSystem
+) -> Result<bool, Box<dyn std::error::Error>> {
+    if !system.get_all_locations().contains(start_state.get_location())
+        {return Err("The transition system does not contain the start location".into())}
+    if !system.get_all_locations().contains(end_state.get_location())
+        {return Err("The transition system does not contain the end location".into())}
+    if let Some(invariants) = end_state.get_location().get_invariants() {
+        if !&end_state.zone_ref().has_intersection(invariants) {
+            return Err("The desired end state is not allowed due to the invariant on this location".into())
+        }
+    }
+
+    Ok(true)
+}
+
+///# Find path
+///
+/// Returns a path from a start state to an end state in a transition system.
+///
+/// If it is reachable, it returns a path.
+///
+/// If it is not reachable, it returns None.
+///
+/// The start state can be omitted with None to use the start state of the transition system.
+///
+///## Checking if a state can reach another:
+///
+/// let is_reachable: bool = match find_path(Some(start_state), end_state, transition_system) {
```

Comment:
Function returns a Result, that contains a options if ok and a string if error
let is_reachable: bool = match find_path(Some(start_state), end_state, transition_system) {
        Ok(option) => match option{
            Some(path) => true,
            None => false,
        },
        Err(string) => panic!(string),
    };